### PR TITLE
feat: presets can be user private

### LINKED ISSUE #640: Create presets for each user
The feature to create presets it's awesome but right now if i create a preset, all users have the same presets. I think it will be great that each user have their own presets. This way each user can choose the alerts they want to see and have their own customize dashboard.

### PR DESCRIPTION
**Pull request by mohanish2504**
/claim #640
/close #640
/split @35C4n0r

### CODE REVIEW

**Comment by 35C4n0r** [comment]
@mohanish2504 kindly add /split @35C4n0r  to the PR body

**Comment by 35C4n0r** [comment]
@shahargl @talboren PR is ready fr review.

**Comment by talboren** [comment]
@mohanish2504 @35C4n0r please add some UI images to the PR 🙏🏼

**Comment by 35C4n0r** [comment]
image
@talboren

**Comment by mohanish2504** [comment]
> image @talboren

@35C4n0r lets match buttons with UI theme?

**Comment by 35C4n0r** [comment]
@talboren @mohanish2504 done
image

**Comment by talboren** [comment]
Modal looks too big. Please decrease its size please

**Comment by 35C4n0r** [comment]
@talboren 
Reduced the size + Matching theme
image

File: keep-ui/app/alerts/alert-presets.tsx [hunk 1]
```diff
@@ -210,6 +219,57 @@ export default function AlertPresets({
     <>
       <Subtitle>Filters</Subtitle>
       <div className="flex w-full">
+        <Modal
+        isOpen={isModalOpen}
+        onClose={() => {}}
```
Comment (talboren) [inline keep-ui/app/alerts/alert-presets.tsx:224]:
```suggestion
        onClose={() => setIsModalOpen(false)}
```

File: keep-ui/app/alerts/alert-presets.tsx [hunk 2]
```diff
@@ -168,11 +173,14 @@ export default function AlertPresets({
   }
 
   async function addOrUpdatePreset() {
-    const newPresetName = prompt(
-      `${preset?.name ? "Update preset name?" : "Enter new preset name"}`,
-      preset?.name === "feed" || preset?.name === "deleted" ? "" : preset?.name
-    );
-    if (newPresetName) {
+    // console.log("HELLO WORLD");
```
Comment (talboren) [inline keep-ui/app/alerts/alert-presets.tsx:176]: Remove comments

File: keep-ui/app/alerts/alert-presets.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-import { Button, Subtitle } from "@tremor/react";
+import Modal from '@/components/ui/Modal';
+import {Button, Subtitle, TextInput} from "@tremor/react";
 import { CheckIcon, PlusIcon, TrashIcon } from "@radix-ui/react-icons";
 import { getApiURL } from "utils/apiUrl";
 import { toast } from "react-toastify";
 import { useSession } from "next-auth/react";
 import { usePresets } from "utils/hooks/usePresets";
 import { useRouter } from "next/navigation";
+import AlertTableCheckbox from "./alert-table-checkbox";
```
Comment (talboren) [inline keep-ui/app/alerts/alert-presets.tsx:13]: What for?

File: keep/api/models/db/preset.py [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
 from sqlmodel import JSON, Column, Field, SQLModel
 
 
 class Preset(SQLModel, table=True):
     # Unique ID for each preset
     id: UUID = Field(default_factory=uuid4, primary_key=True)
+
     tenant_id: str = Field(foreign_key="tenant.id", index=True)
+
+    # if created_by is not null then it means this preset is only for this user
```
Comment (talboren) [inline keep/api/models/db/preset.py:13]: Then why have` is_private` as well?

**Review by talboren — CHANGES_REQUESTED** [review]
Some changes required 🙏🏼

File: keep/api/models/db/preset.py (hunk 4 above)
Comment (35C4n0r) [reply keep/api/models/db/preset.py:13]: @mohanish2504

File: keep-ui/app/alerts/alert-presets.tsx (hunk 2 above)
Comment (35C4n0r) [reply keep-ui/app/alerts/alert-presets.tsx]: done

File: keep-ui/app/alerts/alert-presets.tsx (hunk 3 above)
Comment (35C4n0r) [reply keep-ui/app/alerts/alert-presets.tsx]: resolved

File: keep-ui/app/alerts/alert-presets.tsx (hunk 1 above)
Comment (35C4n0r) [reply keep-ui/app/alerts/alert-presets.tsx]: resolved

File: keep/api/routes/preset.py [hunk 5]
```diff
@@ -22,7 +23,19 @@ def get_presets(
 ) -> list[PresetDto]:
     tenant_id = authenticated_entity.tenant_id
     logger.info("Getting all presets")
-    statement = select(Preset).where(Preset.tenant_id == tenant_id)
+
+    # only global presets
```
Comment (talboren) [inline keep/api/routes/preset.py:27]: This is both global and private, right?

**Review by talboren — CHANGES_REQUESTED** [review]
1 more thing that needs to be done and this can be approved and merged!

In `db.py` search for the `try_create_single_tenant` function and update it to also try and "migrate" the presets table. This is a bit ugly but this is what we got in the meanwhile until we implement a different mechanism.
Look at the example of `TenantApiKey` migration and apply the same for the 2 newly added columns of `Preset`

**Comment by mohanish2504** [comment]
Yep I am aware of it

File: keep/api/core/db.py [hunk 6]
```diff
@@ -193,6 +193,19 @@ def try_create_single_tenant(tenant_id: str) -> None:
...
             pass
 
+    # migrating presets table
+    with Session(engine) as session:
+        try:
+            logger.info("Migrating Preset table")
+            session.exec(
+                "ALTER TABLE preset ADD COLUMN is_private BOOLEAN NOT NULL DEFAULT 0;"
+            )
+            session.exec("ALTER TABLE preset ADD COLUMN created_by TEXT DEFAULT '';")
```
Comment (talboren) [inline keep/api/core/db.py:203]: I'm getting this: `Error Code: 1101. BLOB, TEXT, GEOMETRY or JSON column 'created_by' can't have a default value`

**Review by talboren — CHANGES_REQUESTED** [review]
1 more thing to go 🚀

File: keep/api/core/db.py (hunk 6 above)
Comment (talboren) [reply keep/api/core/db.py:203]:
```suggestion
            session.exec("ALTER TABLE preset ADD COLUMN created_by VARCHAR(1024) DEFAULT '';")
```

**Review by talboren — APPROVED** [review]
LGTM

**Comment by talboren** [comment]
@mohanish2504 @35C4n0r bounty awarded, let me know you got it :) Thanks for the contribution!

**Pull request merged**